Feat 89/accounting backend

PatrickYoussef opened the pull request:
# Description

We can now retrieve the total income and the income for each month. We can also retrieve the total expense and the expense for each month. We can retrieve the top 3 customers based on the total amount they spent.

Issue reference: #89 

## Branch and Commit name must follow this template
```
Branch template: {ISSUE TYPE}-{Issue number}/{desciption_of_branch}
Commit template: {ISSUE TYPE}-{Issue number}: {description of changes}
```

## Type of change

Please delete options that are not relevant.

- [ ] BUG (Fixing or documenting a bug)
- [x] FEAT (Add a new feature to the project)
- [ ] CHORE (Something that needs to be done outside of the issue)
- [ ] DOCS (Implement documentation)
- [ ] TASK (A task that must be done to complete an issue)

# Checklist:

- [x] My pull request references the issue in the comments
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

PatrickYoussef committed:
FEAT-89: total income and expense + blueprint for clients

-get total income and total income per month(and test)
-get total expense and total expense per month(and test)
-what's left: model for top customers

d-rossi committed:
Added get top 3 customers route

- Added getter to obtain the top 3 customers
- Fixed some comments

d-rossi committed:
FEAT-89: Added comments to customer model and service

d-rossi committed:
FEAT-89: Added tests for get top 3 customers route

d-rossi committed:
FEAT-89: Added customers and orders to the db script

PatrickYoussef committed:
FEAT-89: Added more customer orders

PatrickYoussef committed:
Merge branch 'main' into FEAT-89/accounting_backend

PatrickYoussef committed:
FEAT-89: apply lint

PatrickYoussef committed:
Merge branch 'FEAT-89/accounting_backend' of https://github.com/Talos10/SOEN390 into FEAT-89/accounting_backend

phong1233 commented on backend/src/Good/good.service.ts:
```diff
@@ -31,6 +31,57 @@ class Service {
+    /**
+     * Get total expenses
+     */
+    public async getExpense() {
```
Hmm interesting approach but I'm not sure it will work though. Ex: Let's say we have 3 bike at 1000$ each, our expense will be 3000$. What if we sell a bike? The quantity will be 2 which will give us an expense of 2000$ which is not true, I might be wrong but let me know what you think.

PatrickYoussef replied:
Yea, I understand. Once we sell a bike, we lose its expense. Should we replace it with the manufacturing orders?

sebastien-blain replied:
Yes, the expenses should be on the completed manufacturing orders and the income should be on the completed client orders

PatrickYoussef replied:
but, when we order raw goods, for example, those are also expenses, shouldn't we add them ?

sebastien-blain replied:
Manufacturing orders contain raw goods. So they are included.

phong1233 reviewed (commented):
I don't think that calculating the total expense from the goods is the way to go.
We could get it from manufacturing order.

As for the endpoints depending on what we are showing, we could return something that has more data than just the total income / total expense.

Ex: 
```JSON
[
    {
         date: 2015-04-04,
         total_cost: 555
    },
    {
         date: 2015-04-04,
         total_cost: 555
    }
]
```
Using this format the frontend could be setup to display per day/week/month instead of limiting ourselves to per month

PatrickYoussef committed:
Merge branch 'main' into FEAT-89/accounting_backend

PatrickYoussef committed:
FEAT-89: Get expenses from manufacturing

- Now, we get the expenses from manufacturing
- Added data in the DB for relevance

PatrickYoussef committed:
FEAT-89: Apply lint

sebastien-blain reviewed (approved):
Make sure the default value we enter in the database is coherent.

d-rossi committed:
FEAT-89: Updated db values

inventory, manufacturing and order prices are consistent

sebastien-blain committed:
FEAT-89: Fixed manufacturing order and db order
